### PR TITLE
[ECO-5685] Add OpenTok Demo config option for automatic room name generation

### PR DESCRIPTION
**Pull request by emilianop11**

### CODE REVIEW

File: server/serverMethods.js [hunk 1]
```diff
@@ -399,16 +401,18 @@ function ServerMethods(aLogLevel, aModules) {
...
   async function getRoot(aReq, aRes) {
     var meetingAllowed = await isMeetingAllowed(aReq);
     var language = getUserLanguage(accepts(aReq).languages());
     var country = getUserCountry(aReq);
-
+    var roomName = '';
+    
+    if (aReq.tbConfig.autoGenerateRoomName)
+      roomName = `${haikunator.haikunate({ tokenLength: 0 })}-${haikunator.haikunate()}`;
+
```
Comment (Raulkg) [inline server/serverMethods.js:412]: wrap around curly braces {}.

this will add to lint issues

File: web/templates/precall.ejs [hunk 2]
```diff
@@ -24,9 +24,21 @@
...
+          <i data-icon="warning"></i>
+          Please enter a meeting name.
+        </p>
+      </div>
+    </div>
+    <% } %>
     <div id="enter-name-prompt">
       <div class="user-name-input-container">
-        <input id="user-name-input" class="block username" name="userName" autocomplete="off" data-wd="username" autofocus maxlength="46">
+        <input id="user-name-input" class="block username" name="userName" autocomplete="off" data-wd="username" maxlength="46">
```
Comment (Raulkg) [inline web/templates/precall.ejs:41]: is it good idea to validate that username is always present when the auto-generate is turned off ?

```suggestion
        <input id="user-name-input" class="block username" name="userName" autocomplete="off" data-wd="username" maxlength="46" required>
```

Comment (emilianop11) [reply web/templates/precall.ejs:41]: We explicitly allow for blank usernames. They are displayed as "Anonymous user" in the room view

File: web/less/landing.less [hunk 3]
```diff
@@ -170,6 +171,7 @@ a {
       text-shadow: black 1px 1px 12px;
     }
 
+    .room-name-input-container.visited,
```
Comment (jeffswartz) [inline web/less/landing.less:174]: When `autoGenerateRoomName` is set to `false`, in landscape mobile the user name text field is hidden (or offscreen). (Testing in the Chrome mobile emulator.)

File: web/templates/precall.ejs (hunk 2 above)
Comment (Raulkg) [reply web/templates/precall.ejs:41]: alright then it looks fine

**Review by Raulkg — APPROVED** [review]
Except for the linting issue. This looks perfect.

LGTM

File: web/less/landing.less [hunk 4]
```diff
@@ -265,8 +267,7 @@ a {
       position: relative;
       font-size: 16px;
       width: 100%;
-      bottom: 98px;
-      border-bottom: solid 1px @flamingoRed;
+      bottom: 40px;
```
Comment (jeffswartz) [inline web/less/landing.less:270]: I'm not sure what's causing it, but there is a lot of vertical space between the room name text field and the user name text field. I'd say 8px space would be better.

**Review by jeffswartz — CHANGES_REQUESTED** [review]

**Review by jeffswartz — APPROVED** [review]

**Pull request merged**